### PR TITLE
Checks for UUID and sends wallet install event to Posthog

### PR DESCRIPTION
**Pull request by zuuring**
### How to use

Import the Posthog event 
`import { posthogEvent } from "./lib/posthog"`

Call the function where you need events to fire
`posthogEvent("Insert your posthog event title here")`

### Notes
- This adds the basic functionality of Posthog events based on UUID stored from the website.
- if the UUID is already stored as a cookie, send the event to Posthog
- Only "Wallet installed" is setup as an event

### Testing

- [ ] Visit `(https://deploy-preview-113--tally-cash.netlify.app/download/)` to generate a UUID
- [ ] Install the extension for the first time from `dist` of this branch (Delete extension if already installed)
- [ ] Confirm event saved in Posthog `https://app.posthog.com/events`

### CODE REVIEW

File: background/lib/posthog.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      async function fetchCookie(cookie) {
+        if (cookie) {
+          fetch("https://app.posthog.com/capture/", {
+            method: "POST",
+            body: JSON.stringify({
+              api_key: POSTHOG_API_KEY,
+              event: eventName,
+              properties: {
+                distinct_id: cookie.value,
+                data: "This adds posthog events to Tally extension",
```
Comment (0xDaedalus) [inline background/lib/posthog.ts:21]: This seems like it should not be hard coded right?

File: manifest/manifest.json [hunk 2]
```diff
@@ -32,7 +32,14 @@
...
     "default_popup": "popup.html"
   },
-  "permissions": ["alarms", "storage", "unlimitedStorage", "activeTab"],
+  "permissions": [
+    "alarms",
+    "storage",
+    "unlimitedStorage",
+    "activeTab",
+    "cookies",
+    "*://deploy-preview-113--tally-cash.netlify.app/"
```
Comment (0xDaedalus) [inline manifest/manifest.json:41]: I think this is left over from testing

File: background/lib/posthog.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+export function posthogEvent(eventName: string) {
+  if (process.env.SUPPORT_ANALYTICS === "true") {
+    chrome.cookies.get(
+      {
+        url: "https://deploy-preview-113--tally-cash.netlify.app/download/",
+        name: "UUID",
+      },
+
+      async function fetchCookie(cookie) {
```
[the remark this answers is not in the document]
Comment (greg-nagy) [reply background/lib/posthog.ts:12]: it is ... :) it gave me such a hard time :) it's a callback for `chrome.cookies.get` but the extra line is misleading

File: manifest/manifest.json (hunk 2 above)
Comment (greg-nagy) [reply manifest/manifest.json:41]: it's just in the wrong place

File: background/lib/posthog.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// eslint-disable-next-line
+export const POSTHOG_API_KEY = process.env.POSTHOG_API_KEY
+
+export function posthogEvent(eventName: string) {
+  if (process.env.SUPPORT_ANALYTICS === "true") {
```
[the remark this answers is not in the document]
Comment (greg-nagy) [reply background/lib/posthog.ts:5]: 💯

**Comment by greg-nagy** [comment]
@0xDaedalus I will finish this PR, it's not ready for a review yet. Thanks for the stellar comments so far, they have already helped me :)

**Comment by Shadowfiend** [comment]
Closing this as well in favor of the already-merged Posthog/analytics work.

**Pull request closed without merging**